test/tools/yulopti.cpp

```diff
@@ -278,7 +323,22 @@ Allowed options)",
 	string input;
 	try
 	{
-		input = readFileAsString(arguments["input-file"].as<string>());
+		if (arguments.count("input-file"))
```

Comment:
I think it's missing an `else` branch that would report an error if the file is not specified.

Looks like this case wasn't being handled correctly before your PR either. Technically, it will now run as if the file was empty so I guess it's bad but I think an error would be better. Especially for non-interactive use where you might not notice and wonder why you're getting empty output :)

Reply:
It will fall through below in another ``arguments.count("input-file")`` and print the usage message... not that that's nice, but the effect is reasonable :-)... but I can also add an ``else`` here and do the same right away.

Reply:
Ah, right, it's this case that prints help if you don't give you anything. I never liked when programs do that but I can understand why. Personally I'd prefer to have an error (but if anyone really wants the help screen in that case then I would not object that much, it's still reasonable in a way).

Reply:
Well, what it currently does is print the help screen but with failure exit-code (while ``--help`` prints it and indicates success) - maybe that's reasonable :-).

Reply:
sounds like a good compromise :)